doc/sphinx-guides/source/developers/configuration.rst

```diff
@@ -109,3 +109,17 @@ always like ``dataverse.<scope/....>.newname...=old.property.name``. Note this d
 aliases.
 
 Details can be found in ``edu.harvard.iq.dataverse.settings.source.AliasConfigSource``
+
+Adding a Feature Flag
+^^^^^^^^^^^^^^^^^^^^^
+
+Some parts of our codebase might be opt-in only. Experimental or optional feature previews can be switched on using our
```

Comment:
Suggested addition after the first sentence. They may be only for developers/not intended for production, and/or temporary. (Experimental or optional features that can be used in production and are intended to remain, perhaps with some change over time, Experimental or optional feature previews...

Reply:
Actually I think I got what causes the confusion here - we already do experimental things and simply used configuration and not flags for this before!

Here's a bold statement: as we seem to getting more and more of these now, how about we make all the other experimental things using this flag thing, too, so we are consistent again?

Reply:
That would put #9290 which shouldn't get used in production and presumably will be going away back in the list of things you can use now and that we expect to stay around. I suppose that could be handled by strong messaging on that PR.

It also means anyone using these features has to reconfigure for 5.13? And we'd have to decide what to do with features that require some config info - do all their settings move to be under feature or do you turn them on one way and configure them elsewhere? Again, solvable problems, but I don't see a lot of benefit either.

This is definitely something where there probably isn't one right answer and I don't care much how it comes out as long as we don't confuse experimental things you can use in production with things that shouldn't through some form of messaging.

Reply:
> That would put #9290 which shouldn't get used in production and presumably will be going away back in the list of things you can use now and that we expect to stay around. I suppose that could be handled by strong messaging on that PR.

That is a good and valid point - maybe there should be a forth column in the list that indicates security level? Maybe this should even be marked `@Deprecated` in code if it is serious for production so it becomes more visible this needs more work.

> It also means anyone using these features has to reconfigure for 5.13? And we'd have to decide what to do with features that require some config info - do all their settings move to be under feature or do you turn them on one way and configure them elsewhere? Again, solvable problems, but I don't see a lot of benefit either.

Well if we want to migrate existing things to this mechanism, we certainly will be able to make this backward compatible. (As done before for JvmSettings)
 
> This is definitely something where there probably isn't one right answer and I don't care much how it comes out as long as we don't confuse experimental things you can use in production with things that shouldn't through some form of messaging.

I agree. Another thought: while the session auth thing is maybe not perfectly secure, isn't this something that we probably will turn on during the process of moving to the SPA at least at demo.dataverse.org? We will need that session auth thing if we want to demonstrate using both in parallel. This kind of makes this thing a bit blury.

Reply:
I think the existing line conveys the main points:

- opt-in
- experimental
- optional

That said, no objection if someone wants to add more. To me, feature flags are a common term in the industry. Most developers have probably heard of them.